### PR TITLE
added-snyk

### PR DESCRIPTION
**Pull request by mayank0202**
Please review `kms-key-arn` . Snyk wanted to add this attribute and the value to this attribute should be customer managed key

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -271,7 +274,7 @@ resource "aws_dynamodb_table" "terraform_state_lock" {
   }
 
   point_in_time_recovery {
-    enabled = var.enable_dynamodb_point_in_time_recovery
+    enabled = true
```
Comment (jamescrowley321) [inline main.tf]: Make this a variable.

File: variables.tf [hunk 2]
```diff
@@ -6,11 +6,11 @@ variable "dynamodb_name" {
   description = "The name of the table, this needs to be unique within a region."
 }
 
-variable "enable_dynamodb_point_in_time_recovery" {
-  description = "Whether to enable point-in-time recovery - note that it can take up to 10 minutes to enable for new tables."
-  type        = bool
-  default     = false
```
Comment (jamescrowley321) [inline variables.tf:12]: You can just make the default true.

**Review by jamescrowley321 — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (mayank0202) [reply variables.tf:12]: done

File: main.tf (hunk 1 above)
Comment (mayank0202) [reply main.tf]: done

File: main.tf [hunk 3]
```diff
@@ -88,8 +88,10 @@ resource "aws_s3_bucket" "private" {
     Name = var.bucket_name,
   }))
 
+  logging {}
   versioning {
     enabled = var.enable_versioning
+    mfa_delete = true
```
Comment (jamescrowley321) [inline main.tf]: Needs to be a variable and set to `false`. You can't just turn on MFA settings like this. There is additional upstream configuration that is required.

File: main.tf [hunk 4]
```diff
@@ -263,6 +265,7 @@ resource "aws_dynamodb_table" "terraform_state_lock" {
 
   server_side_encryption {
     enabled = true
+    kms_key_arn = data.aws_iam_policy_document.policy.arn
```
Comment (jamescrowley321) [inline main.tf]: You are pointing to the incorrect resource here. You should also be able to specify the default KMS key for dynamo. It should have an alias like `aws/dynamo`

**Review by jamescrowley321 — CHANGES_REQUESTED** [review]

File: main.tf (hunk 3 above)
Comment (mayank0202) [reply main.tf]: done

File: main.tf (hunk 4 above)
Comment (mayank0202) [reply main.tf]: done

File: variables.tf [hunk 5]
```diff
@@ -2,22 +2,28 @@
...
 variable "dynamodb_hash_key" {
-  type = string
+  type        = string
   description = "The attribute to use as the hash (partition) key."
   default     = "LockID"
 }
 
+variable "dynamo_kms_master_key_id" {
+  type        = string
+  description = "The ID of an AWS-managed customer master key (CMK) for Amazon SNS or a custom CMK"
```
Comment (jamescrowley321) [inline variables.tf]: We dont want to specify a CMK. This also mentions SNS.

File: main.tf [hunk 6]
```diff
@@ -262,7 +264,8 @@ resource "aws_dynamodb_table" "terraform_state_lock" {
   write_capacity = 2
 
   server_side_encryption {
-    enabled = true
+    enabled     = true
+    kms_key_arn = local.dynamo_kms_master_key_id
```
Comment (jamescrowley321) [inline main.tf:268]: Leave out this parameter if you can for now.

**Review by jamescrowley321 — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 5 above)
Comment (mayank0202) [reply variables.tf]: Nope it doesnt mention sns .. Actually i copied the description so forgot to change it

File: main.tf (hunk 6 above)
Comment (mayank0202) [reply main.tf:268]: Then we need to add this in ignore file .. I added in locals like this 
` dynamo_kms_master_key_id = var.dynamo_kms_master_key_id == null ? "alias/aws/dynamo" : var.dynamo_kms_master_key_id`

Comment (mayank0202) [reply main.tf:268]: can we go ahead with this? or need to remove this parameter? @jamescrowley321

File: variables.tf (hunk 5 above)
Comment (jamescrowley321) [reply variables.tf]: Just update the description and you are good.

File: main.tf (hunk 6 above)
Comment (jamescrowley321) [reply main.tf:268]: You are good to go. I missed the local.

File: variables.tf (hunk 5 above)
Comment (mayank0202) [reply variables.tf]: done

**Review by jamescrowley321 — APPROVED** [review]

**Pull request merged**